### PR TITLE
Add Heat Bomb Launcher

### PR DESCRIPTION
**Pull request by pscamman**
Hey everyone! I've just been getting started tinkering with this repo. Very exciting. I figured I'd throw a very simple pull request up just to start getting feedback.

### In-game description
Heat Bombs are made with a special material that absorbs heat phenomenally well. Unfortunately, for those interested in designing dedicated cooling systems, it also becomes severely unstable when superheated. Faced with this challenge, one clever engineer decided it would be best to simply eject the material from the ship. In the direction of an enemy.

### More details
This is basically a shorter range, extra strong Heavy Rocket with heat damage and the added twist of reducing your heat rather than adding to it. I added a simple check in the C++ code to prevent weapons from firing if they would drop your heat below 0 (I believe this is the first weapon to deduct heat).

I don't know anything about creating art assets, but if that would be expected of me then I'd be more than glad to learn how to do all that. For now this weapon just borrows assets from the Torpedo.

For now, I added the weapon to outfitters that carry Plasma Cannons. I did not add it to any ships.

Lastly, I added one line to the .gitignore to accommodate my IDE, and changed a few words in the developer readme to make a sentence more clear. If I should really separate those changes into a different pull request, I could easily do that.

### Weapon stat highlights
```
cost 200000
"outfit space" -35
"weapon capacity" -35
"heat bomb capacity" 20
"velocity" 10
"lifetime" 50
"reload" 120
"firing heat" -1000
"trigger radius" 30
"blast radius" 50
"shield damage" 900
"hull damage" 900
"heat damage" 500
"hit force" 800
"missile strength" 35
```

### CODE REVIEW

File: data/weapons.txt [hunk 1]
```diff
@@ -846,6 +846,62 @@ effect "torpedo fire"
...
+	description "Heat Bombs are ammunition for the Heat Bomb Launcher. Each Heat Bomb Launcher can hold 20 of them. You cannot install Heat Bombs unless you have also installed a launcher to fire them from."
+
+outfit "Heat Bomb Storage Rack"
+	category "Ammunition"
+	cost 60000
+	thumbnail "outfit/torpedo storage"
+	"mass" 3
+	"outfit space" -9
+	"torpedo capacity" 10
+	ammo "Torpedo"
```
Comment (Amazinite) [inline data/weapons.txt]: torpedo capacity —> heat bomb capacity
Torpedo —> Heat Bomb
Don’t want to be storing the wrong ammo.

**Comment by pscamman** [comment]
Ok, cool! Thanks for all the tips!

**Comment by pscamman** [comment]
Closing so I can redo my version control to work better with the suggestions.

**Comment by tehhowch** [comment]
Am I alone in feeling that a heat bomb is a very "the Korath are innovating again" weapon?  Obviously needs a nicer name but sure is in line with that lore bit about the flames shooting out of the ship being intentional. 👍

**Comment by pscamman** [comment]
I thought a bit about the Korath with this one as well. It would probably
be too complicated to work well, but I imagined Korath Bomber fighters that
would absorb heat from their parent ship while docked and then use it to
launch these bombs.

**Pull request closed without merging**